storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -103,3 +103,52 @@ func (d *DotGit) Idxfile() (fs.FS, string, error) {
 
 	return nil, "", ErrIdxNotFound
 }
+
+func (d *DotGit) Objectfiles() (fs.FS, []core.Hash, error) {
+	dotGitobjcts := d.fs.Join(d.path, "objects")
+	files, err := d.fs.ReadDir(dotGitobjcts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var objDirs []string
+	reDir, _ := regexp.Compile("[a-z0-9]{2}")
```

Comment:
Use regexp.MustCompile instead of regexp.Compile.

Also move the resulting regexp to a package level variable an initialize it in the package init.